src/nbl/video/CVulkanDescriptorPool.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __NBL_C_VULKAN_DESCRIPTOR_POOL_H_INCLUDED__
+
+#include "nbl/video/IDescriptorPool.h"
+
+#include <volk.h>
+
+namespace nbl::video
+{
+
+class ILogicalDevice;
+
+class CVulkanDescriptorPool : public IDescriptorPool
+{
+public:
+    CVulkanDescriptorPool(core::smart_refctd_ptr<ILogicalDevice>&& dev, VkDescriptorPool descriptorPool)
+        : IDescriptorPool(std::move(dev)), m_descriptorPool(descriptorPool)
+    {}
+
+    ~CVulkanDescriptorPool();
+
+    inline VkDescriptorPool getInternalObject() const { return m_descriptorPool; }
+
+private:
```

Comment:
remind me, when you create a Vulkan descriptor pool, do you specify a max descriptor count?

Reply:
no

Reply:
wrong theere are counters for max descriptors of each type.